mysql/assets/configuration/spec.yaml

```diff
@@ -0,0 +1,319 @@
+name: MySQL
+files:
+  - name: mysql.yaml
+    options:
+      - template: init_config
+        options:
+          - name: global_custom_queries
+            description: |
+              Global custom queries can be applied to all instances using the
+              `use_global_custom_queries` setting at the instance level.
+
+              See `custom_queries` defined below for syntax.
+            value:
+              type: object
+
+          - template: init_config/default
+
+      - template: instances
+        options:
+          - name: server
+            required: true
+            description: |
+              MySQL server to connect to.
+              NOTE: Even if the server name is "localhost", the agent connects to MySQL using TCP/IP, unless you also
+              provide a value for the sock key (below).
+            value:
+              type: string
+              example: localhost
+
+          - name: user
+            required: true
+            description: |
+              Datadog username created to connect to MySQL.
+            value:
+              type: string
+
+          - name: pass
+            required: true
+            description: |
+              Password associated to the Datadog user.
+            value:
+              type: string
+
+          - name: port
+            description: |
+              Port to use when connecting to MySQL.
+            value:
+              type: number
+              example: 3306
+
+          - name: sock
+            description: |
+              Path to a Unix Domain Socket to use when connecting to MySQL (instead of a TCP socket).
+            value:
+              type: string
+
+          - name: defaults_file
+            description: |
+              Path to an alternative configuration mechanism file.
+            value:
+              type: string
+
+          - name: connect_timeout
+            description: |
+              Maximum number of seconds to wait before timing out when connecting to MySQL.
+            value:
+              type: number
+              example: 10
+
+          - name: ssl
+            description: |
+              Use this section to configure a TLS connection between the Agent and MySQL.
+            options:
```

Comment:
I think you need to define its value as object and then the properties inside it

Reply:
Doesn't look like `properties` actually does anything, eg see https://github.com/DataDog/integrations-core/blob/74e71bad320547a9518dcdb3f6f4bea1ecfebc41/mongo/assets/configuration/spec.yaml#L143-L177 vs https://github.com/DataDog/integrations-core/blob/74e71bad320547a9518dcdb3f6f4bea1ecfebc41/mongo/datadog_checks/mongo/data/conf.yaml.example#L139-L152

I went with an `example` for now, same as before.